### ISSUE TITLE
feat: Migrate audio processing from PyTorch NAM to Pedalboard + NAM VST3

### ISSUE THREAD

**Issue by krazyuniks** [krazyuniks/guitar-tone-shootout#56]
## Summary

Migrate the audio processing pipeline from the current PyTorch-based `neural-amp-modeler` library to Spotify's **Pedalboard** with the **NAM VST3 plugin**. This enables:

- **Parallel signal chains** via `pedalboard.Mix()` for true A/B comparison processing
- **Unified audio processing** - all effects (NAM, IR, EQ, reverb) in one Pedalboard chain
- **C++ performance** instead of Python/PyTorch overhead
- **Volume normalization** built into the processing pipeline

## Background

A proof-of-concept in `/Users/ryanlauterbach/Work/pedalboard_nam_poc/` successfully demonstrated:

1. **Programmatic VST3 preset generation** - We can dynamically create `.vstpreset` files with embedded model paths
2. **NAM model loading** - The NAM VST3 plugin loads models from presets without GUI interaction
3. **Full signal chain** - NAM → IR Convolution → Effects all in Pedalboard
4. **Volume normalization** - Input (-18 dB RMS) and output (-14 dB RMS) normalization

### Key Discovery: VST3 Preset Format

The NAM VST3 plugin stores the model path in its component state:

```
VST3 Header (48 bytes):
├── Magic: "VST3"
├── Version: 1
├── Class ID: "F2AEE70D00DE4F4E534441613159456F"
└── Chunk list offset

NAM Component State (iPlug2 format):
├── String: "###NeuralAmpModeler###"
├── String: "0.7.13" (version)
├── String: "/path/to/model.nam"  ← Model path!
├── String: "/path/to/ir.wav" (optional, we use Pedalboard's IR)
└── Parameters: 12 doubles (normalized 0-1 values)
```

### 1. Create Preset Generation Module
- [ ] Extract `generate_nam_preset()` and related functions from PoC
- [ ] Create `pipeline/src/guitar_tone_shootout/preset.py`
- [ ] Add unit tests for preset generation
- [ ] Support both NAM-only and NAM+IR presets

### 2. Refactor Audio Processing (`audio.py`)
- [ ] Replace PyTorch NAM processing with Pedalboard + VST3
- [ ] Implement `load_nam_via_preset()` function
- [ ] Keep IR convolution via `pedalboard.Convolution`
- [ ] Maintain existing effect presets (EQ, reverb, delay)
- [ ] Add input normalization (target: -18 dB RMS)
- [ ] Add output normalization (target: -14 dB RMS)
- [ ] Update `process_chain()` to use unified Pedalboard chain

### 3. Add Volume Normalization Module
- [ ] Create `pipeline/src/guitar_tone_shootout/normalize.py`
- [ ] Implement `normalize_rms(audio, target_db)` function
- [ ] Implement `rms_db(audio)` measurement function
- [ ] Add peak limiting to prevent clipping
- [ ] Consider LUFS normalization for future (broadcast standard)

### 4. Docker Integration
- [ ] Add NAM VST3 plugin to worker container
- [ ] Configure VST3 plugin paths in container
- [ ] Test headless operation (no GUI)
- [ ] Update `Dockerfile.dev` and `Dockerfile`

### 5. Update Dependencies
- [ ] Remove `neural-amp-modeler` from `pyproject.toml`
- [ ] Ensure `pedalboard>=0.9.0` is present
- [ ] Remove PyTorch dependency (unless needed elsewhere)
- [ ] Update lockfile

### 6. Update Documentation
- [x] Add comparison methodology to README.md
- [ ] Update `AGENTS.md` with new audio processing patterns
- [ ] Document VST3 preset format in dev docs
- [ ] Update pipeline skill documentation

### 7. Testing
- [ ] Unit tests for preset generation
- [ ] Integration tests for full signal chain
- [ ] Compare output quality vs old PyTorch implementation
- [ ] Performance benchmarks (processing time)

## New Architecture

```python
# Before (current)
from nam.models import init_from_nam
model = init_from_nam(config)
output = model(torch.tensor(audio))  # PyTorch

# After (new)
from pedalboard import Pedalboard, Convolution, load_plugin
from guitar_tone_shootout.preset import generate_nam_preset

# Generate preset with model path
generate_nam_preset(model_path, preset_path)

# Load via Pedalboard
nam = load_plugin("NeuralAmpModeler.vst3")
nam.load_preset(preset_path)
nam.irtoggle = False  # Use Pedalboard's IR

# Full chain
chain = Pedalboard([
    Gain(gain_db=input_gain),  # Input normalization
    nam,
    Convolution(ir_path),
    # ... effects ...
    Gain(gain_db=output_gain),  # Output normalization
])
output = chain(audio, sample_rate)
```

## Benefits

| Aspect | Before (PyTorch) | After (Pedalboard) |
|--------|------------------|-------------------|
| Parallel chains | ❌ Sequential only | ✅ `Mix()` for parallel |
| Performance | Python + PyTorch | C++ native |
| Dependencies | PyTorch (~2GB) | Pedalboard (~50MB) |
| Processing API | Split (NAM + Pedalboard) | Unified |
| Model loading | Direct file load | Preset with path |

## Files to Modify

```
pipeline/
├── src/guitar_tone_shootout/
│   ├── audio.py           # Major refactor
│   ├── preset.py          # NEW - preset generation
│   ├── normalize.py       # NEW - volume normalization
│   └── pipeline.py        # Update chain processing
├── pyproject.toml         # Update dependencies
└── tests/
    ├── test_preset.py     # NEW
    └── test_normalize.py  # NEW

docker/
├── Dockerfile.dev         # Add NAM VST3
└── Dockerfile             # Add NAM VST3

dev/
└── active/pedalboard-migration/  # Task documentation
```

## PoC Reference

The working proof-of-concept is at:
```
~/Work/pedalboard_nam_poc/
├── SUMMARY.md              # Full documentation
├── poc.py                  # Working code
├── jcm800.vstpreset        # Generated preset example
├── output_nam_plus_ir.wav  # NAM + IR output
└── normalized_*.wav        # Normalized outputs
```

## Acceptance Criteria

- [ ] All existing pipeline tests pass
- [ ] NAM models load via generated presets
- [ ] IR convolution works via Pedalboard
- [ ] Volume normalization produces consistent output levels
- [ ] Docker worker can process audio headlessly
- [ ] No regression in output audio quality
- [ ] Performance equal or better than PyTorch implementation

## Related

- PoC: `~/Work/pedalboard_nam_poc/SUMMARY.md`
- Pedalboard: https://github.com/spotify/pedalboard
- NAM Plugin: https://github.com/sdatkinson/NeuralAmpModelerPlugin
- VST3 Preset Format: https://steinbergmedia.github.io/vst3_dev_portal/pages/Technical+Documentation/Locations+Format/Preset+Format.html
